MVP Module/Source/Feed UI/Controllers/FeedRefreshViewController.swift

```
@@ -0,0 +1,41 @@
+
+import UIKit
+import FeedFeature
+
+protocol FeedRefreshControllerDelegate {
+    func didRequestFeedRefresh()
+}
+
+final class FeedRefreshController: NSObject, FeedRefreshView {
+    
+    @IBOutlet var view: UIRefreshControl?
+    @IBOutlet var errorView: ErrorView?
+    
+    var delegate: FeedRefreshControllerDelegate?
+    
+    @IBAction func refresh() {
+        errorView?.hideMessage()
+        delegate?.didRequestFeedRefresh()
+    }
+    
+    func display(_ viewModel: FeedLoadingViewModel) {
+        displayRefreshLoader(viewModel.isLoading)
+        
+        if let errorMessage = viewModel.errorMessage {
+            displayErrorMessage(errorMessage)
+        }
```

It's missing an else case here, where you hide the error message for the `nil` case: `errorView?.hideMessage()`

This way, the controller doesn't decide when to show/hide the error view. It just obeys to commands it receives via the presenter.